iroha_permissions_validators/src/lib.rs

```diff
@@ -2,15 +2,101 @@
 
 #![allow(clippy::module_name_repetitions)]
 
-use std::convert::TryInto;
+use std::{collections::BTreeMap, convert::TryInto};
 
 use iroha::{
     expression::Evaluate,
-    permissions::{prelude::*, PermissionsValidator, PermissionsValidatorBuilder},
+    permissions::{
+        prelude::*, GrantedTokenValidator, PermissionsValidator, PermissionsValidatorBuilder,
+        ValidatorApplyOr,
+    },
     prelude::*,
 };
 use iroha_data_model::{isi::*, prelude::*};
 
+macro_rules! impl_from_item_for_validator_box {
+    ( $ty:ty ) => {
+        impl From<$ty> for PermissionsValidatorBox {
+            fn from(permissions: $ty) -> Self {
+                Box::new(permissions)
```

Comment:
May be it would be better to implement it as an iterator? Generate it for all input arguments and have all this from implemented in single macro call?